opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/bond_encoder.rs

```diff
@@ -34,28 +34,72 @@ pub(crate) enum BondDataType {
 /// Bond protocol writer for encoding values to buffers
 pub(crate) struct BondWriter;
 
+// Trait for types that can be converted to little-endian bytes
+/// This is automatically implemented for all primitive numeric types
+trait ToLeBytes {
+    type ByteArray: AsRef<[u8]>;
+    fn to_le_bytes(self) -> Self::ByteArray;
+}
+
+// Implement for all standard numeric types
+macro_rules! impl_to_le_bytes {
+    ($($t:ty),*) => {
+        $(
+            impl ToLeBytes for $t {
+                type ByteArray = [u8; std::mem::size_of::<$t>()];
+                fn to_le_bytes(self) -> Self::ByteArray {
+                    <$t>::to_le_bytes(self)
+                }
+            }
+        )*
+    };
+}
+
+impl_to_le_bytes!(i8, i16, i32, i64, u8, u16, u32, u64, f32, f64);
```

Comment:
Do we need to implement it for all these types? If not, I suggest implementing this only for the types that we would work with.

Reply:
It looks like you have it for `u32` as well.

Reply:
trace flags need to be `u32`. Will fix this in subsequent PR - https://github.com/open-telemetry/opentelemetry-rust/blob/8d46c40b60ad95135e63316f87edc7ecc9ae45b1/opentelemetry-proto/src/proto/tonic/opentelemetry.proto.logs.v1.rs#L141

Reply:
#355 to track.